hs_core/hydroshare/resource.py

```diff
@@ -957,17 +957,20 @@ def delete_format_metadata_after_delete_file(resource, file_name):
             resource.metadata.delete_element(format_element.term, format_element.id)
 
 
-def delete_resource_file(pk, filename_or_id, user):
+def delete_resource_file(pk, filename_or_id, user, delete_logical_file=True):
     """
-    Deletes an individual file from a HydroShare resource. If the file does not exist, the Exceptions.NotFound exception
-    is raised.
+    Deletes an individual file from a HydroShare resource. If the file does not exist,
+    the Exceptions.NotFound exception is raised.
 
     REST URL:  DELETE /resource/{pid}/files/{filename}
 
     Parameters:
     pk - The unique HydroShare identifier for the resource from which the file will be deleted
     filename - Name of the file to be deleted from the resource
     user - requesting user
+    delete_logical_file - If True then the ResourceFile object to be deleted if it is part of a
```

Comment:
@alvacouch 

> I would like to have one concept of logical file in the system. Also, and this is more important, we should not be deleting logical files via side effects. The proper way to delete a logical file is by following a relation from ResourceFile to the LogicalFile for which it is a leaf, and then delete the LogicalFile.....

My understanding is that deleting a file that is part of a logical file makes the file type invalid - no more in usable state. That's the reason we delete all files when one of the files is deleted. That's the logic currently implemented in many existing resource types ( e.g time series, geographic feature, raster etc).

Since the UI currently does not have functionality to delete a logical file, hence there is not a view function for that. However, deletion of logical file has been implemented at the internal api level. So in the future when the UI is ready for logical file delete, it can be easy supported at the back-end.